File: OneSignalSDK/onesignal/src/main/java/com/onesignal/FCMIntentJobService.java
package com.onesignal;

import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;

import static com.onesignal.NotificationBundleProcessor.processBundleFromReceiver;

/**
 * Uses modified JobIntentService class that's part of the onesignal package
 */
@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
public class FCMIntentJobService extends JobIntentService {

    public static final String BUNDLE_EXTRA = "Bundle:Parcelable:Extras";
    private static final int JOB_ID = 123890;

    @Override
    protected void onHandleWork(@NonNull Intent intent) {
        Bundle bundle = intent.getExtras();
        if (bundle == null)
            return;

        OneSignal.initWithContext(this);
        processBundleFromReceiver(this, bundle, new NotificationBundleProcessor.ProcessBundleReceiverCallback() {
            @Override
            public void onBundleProcessed(@Nullable NotificationBundleProcessor.ProcessedBundleResult processedResult) {

            }
        });
    }

    public static void enqueueWork(Context context, Intent intent) {
        enqueueWork(context, FCMIntentJobService.class, JOB_ID, intent, false);
    }
}

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/onesignal/device/services/FCMIntentJobService.kt
package com.onesignal.onesignal.device.services

import android.content.Context
import android.content.Intent
import android.os.Build
import androidx.annotation.RequiresApi
import com.onesignal.onesignal.OneSignal
import com.onesignal.onesignal.internal.notification.work.NotificationBundleProcessor

/**
 * Uses modified JobIntentService class that's part of the onesignal package
 */
@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
class FCMIntentJobService : JobIntentService() {
    protected override fun onHandleWork(intent: Intent) {
        val bundle = intent.extras ?: return
        OneSignal.initWithContext(this)
        val bundleProcessor = OneSignal.getService<NotificationBundleProcessor>()
        bundleProcessor.processBundleFromReceiver(
            this,
            bundle,
            object : NotificationBundleProcessor.ProcessBundleReceiverCallback {
                override fun onBundleProcessed(processedResult: NotificationBundleProcessor.ProcessedBundleResult?) {}
            })
    }

    companion object {
        const val BUNDLE_EXTRA = "Bundle:Parcelable:Extras"
        private const val JOB_ID = 123890
        fun enqueueWork(context: Context?, intent: Intent?) {
            enqueueWork(context!!, FCMIntentJobService::class.java, JOB_ID, intent!!, false)
        }
    }
}